test/ref_ops_test.cpp

```diff
@@ -7188,6 +7188,68 @@ TEST_CASE(slice_test)
     }
 }
 
+TEST_CASE(slice_dyn_test)
+{
+    {
+        // Slice a single dynamic dimension
+        migraphx::program p;
+        auto* mm = p.get_main_module();
+        std::vector<int> data(2 * 2 * 3);
+        std::iota(data.begin(), data.end(), 0);
+        migraphx::shape s{migraphx::shape::int32_type, {{2, 2, 0}, {2, 2, 0}, {3, 3, 0}}};
```

Comment:
This is fixed dynamic shape. Can you make dims other than axes dynamic ?

Reply:
done.  Makes the test more comprehensive.